### ISSUE TITLE
Nested list item not rendered in inside list

### ISSUE THREAD

**Issue by jimmyvdberg** [aurelia/templating#562]
**I'm submitting a bug report**

* **Aurelia CLI version:**
0.31.3

**Please tell us about your environment:**
* **Operating System:**
Windows 10

* **Node Version:**
7.7.1

* **NPM Version:**
5.4.1

* **Browser:**
all

* **Language:**
TypeScript 2.5.2

**Current behavior:**
I'm having some issues with rendering a nested list (`<ul>`) within a custom element and a slot.
The inner list item (`<li>`) is not rendered in the inner list (`<ul>`) but in the outside list...

**Steps to reproduce:**
- Create a new Aurelia project with the CLI (`au new`) 
- Create a custom element called "sample-element.html" in the resources/elements folder. With the following content:
```
<template>
  <ul>
    <slot name="items">
      <li>ITEMS PLACHOLDER</li>
    </slot>
  </ul>
</template>
```
- Replace the contents of the app.html file with:
```
<template>
  <require from="resources/elements/sample-element.html"></require>
  <ul>
    <li>
      HOME
    </li>
    <li>
      <sample-element>
        <li slot="items">Item 1</li>
      </sample-element>
    </li>
  </ul>
</template>
```

I also created this gist with the files created/replaced: https://gist.run/?id=07b4cb00f32d059f1d16957137511e55

**Expected/desired behavior:**
What I expect to see is that `<li>Item 1</li>` is rendered inside the nested list. But instead the placeholder of the slot is rendered inside that list. Item 1 is rendered even outside my custom `sample-element`.
Issue:
![image](https://user-images.githubusercontent.com/1083192/30483979-2afa59de-9a29-11e7-9012-6aa10c89c527.png)
Expected:
![image](https://user-images.githubusercontent.com/1083192/30484026-59bda136-9a29-11e7-8615-d4b37bea1928.png)

When I change the list item tag to a random other tag it is working as expected.
```
<sample-element>
  <span slot="items">Item 1</span>
</sample-element>
```
![image](https://user-images.githubusercontent.com/1083192/30484153-c9f16dd4-9a29-11e7-8b83-e56c320b5deb.png)

Can someone help solving this issue? :-)

**Comment by bigopon** [comment]
It is because Chrome automatically relocates nested `li` next to the nesting `li` before Aurelia inspect the template. `li` is not a valid direct child of `li`.

You can test it by doing this in your console:

```js
var div = document.createElement('div');
div.innerHTML = `
  <ul>
    <li>
      1
      <li>2</li>
    </li>
  </ul>
`
```

The actual tree you will get (in Chrome)

```html
<div>
  <ul>
    <li>1</li>
    <li>2</li>
  </ul>
</div>
```

**Comment by jimmyvdberg** [comment]
Aaaa thanks! Shouldn't Aurelia inspect the template sooner then? I understand why Chrome is doing this but if you write this code I still expect it to work.

Do you have a suggestion on how to make this work?

**Comment by bigopon** [comment]
Aurelia inspect the template element, not the template string. In order to have template element, it needs the browser to parse template string into a template element, that's where invalid markup gets removed / adjusted like above. I don't know and easy way to make it work.

You can implement a custom view strategy, where you can pass virtual dom like object to construct the template element:

```js
@virtualDomViewStrategy({
  children: [
    { el: 'li', children: [
      {  el: 'li' }
    ]}
  ]
})
```

You can have a look at the `inlineView`, `useView` decorators in `decorators.js` of this repo to know where to start.

**Comment by jimmyvdberg** [comment]
Thanks for the help. I created another custom element with only a `li` in it. And used that custom element inside the `ul`.

```
<template>
  <li>
    <slot></slot>
  </li>
</template>
```

app.html:
```
<sample-element>
 <sample-item slot="items">Item 1</sample-item>
 <sample-item slot="items">Item 2</sample-item>
</sample-element>
```
It now has the expected results:
![image](https://user-images.githubusercontent.com/1083192/30531366-21ca582e-9c4e-11e7-8798-095a33d6f945.png)

I also made a new Gist with updated code: https://gist.run/?id=37b8d3aa7898b38f63956c0d0677f707
